### PR TITLE
only check for psql in config.php

### LINKED ISSUE #1760: Update script should check if postgresql is used before updateing
If someone updates the VM on a server where Nextcloud-VM is driven by  Postgres-SQL and a MySQL-Server is installed the update-Script currently shows the message below:

### Message to surpress if PostgreSQL is used

Bild_2021-01-10_120636

This may be intended to make money or just as a warning that Mysql is not supported anymore. 

But If Postgres and MySQL are installed side by side this is a bug. 

So there should be a function to check which database driver is used in settings.php to avoid that message if it is not needed

That happens on Ubuntu 20.04 when the script is used to update to Nextcloud 20.04 Of course there is a way to do this manually by using the nextcloud-Updater. 

But the message itself is somewhat annoing.

### dpkg -l output

[pckglist.txt](https://github.com/nextcloud/vm/files/5792513/pckglist.txt)

##### Line 729 - 734
```bash
mariadb-client-10.3                  1:10.3.25-0ubuntu0.20.04.1                 amd64        MariaDB database client binaries
ii  mariadb-client-core-10.3             1:10.3.25-0ubuntu0.20.04.1                 amd64        MariaDB database core client binaries
ii  mariadb-common                       1:10.3.25-0ubuntu0.20.04.1                 all          MariaDB common metapackage
ii  mariadb-server                       1:10.3.25-0ubuntu0.20.04.1                 all          MariaDB database server (metapackage depending on the latest version)
ii  mariadb-server-10.3                  1:10.3.25-0ubuntu0.20.04.1                 amd64        MariaDB database server binaries
ii  mariadb-server-core-10.3             1:10.3.25-0ubuntu0.20.04.1                 amd64        MariaDB database core server files
```

##### Line 1094 - 1098
```bash
ii  postgresql                           12+214ubuntu0.1                            all          object-relational SQL database (supported version)
ii  postgresql-12                        12.5-0ubuntu0.20.04.1                      amd64        object-relational SQL database, version 12 server
ii  postgresql-client-12                 12.5-0ubuntu0.20.04.1                      amd64        front-end programs for PostgreSQL 12
ii  postgresql-client-common             214ubuntu0.1                               all          manager for multiple PostgreSQL client versions
ii  postgresql-common                    214ubuntu0.1                               all          PostgreSQL database-cluster manager
```

### Some more system information

````bash
          .-/+oossssoo+/-.                
        `:+ssssssssssssssssss+:`           -------------------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 20.04.1 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: KVM/QEMU (Standard PC (i440FX + PIIX, 1996) pc-i440fx-2.11) 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.4.0-60-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 1 day, 12 hours, 38 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 1224 (dpkg), 4 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.0.17 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Terminal: /dev/pts/0 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   CPU: Intel Xeon E5-2630 v4 (4) @ 2.199GHz 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   GPU: 00:02.0 Vendor 1234 Device 1111 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Memory: 6090MiB / 11981MiB 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/                            
  +sssssssssdmydMMMMMMMMddddyssssssss+                             
   /ssssssssssshdmNNNNmyNMMMMhssssss/
    .ossssssssssssssssssdMMMNysssso.
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.
````
### Corrosponding section in config.php to check:

```php
 ),
  'datadirectory' => 'xxx',
  'dbtype' => 'pgsql',
  'version' => '20.0.4.0',
  'overwrite.cli.url' => 'xxx',
  'dbname' => 'nextcloud_db',
  'dbhost' => 'localhost',
  'dbport' => '',

```

### PR DESCRIPTION
**Pull request by enoch85**
Fixes #1760

This also reverts https://github.com/nextcloud/vm/commit/282cbe7e4aa9861ef59c3982000ad79c07e65aff#diff-67bceb0888b398c45857e1bc41f753da1a696075ff08b9b82f8136bc6af0139a (https://github.com/nextcloud/vm/issues/1705)

### CODE REVIEW

File: nextcloud_update.sh [hunk 1]
```diff
@@ -108,7 +108,7 @@ Holding PHP to avoid upgrading to a newer version without migration..."
 fi
 
 # Don't allow MySQL/MariaDB
-if ! grep -q pgsql /var/www/nextcloud/config/config.php || dpkg -l | grep -q mariadb-common
+if ! grep -q pgsql /var/www/nextcloud/config/config.php && dpkg -l | grep -q mariadb-common
```
Comment (szaimen) [inline nextcloud_update.sh:111]: @enoch85 I guess this is how it should work

File: nextcloud_update.sh [hunk 2]
```diff
@@ -106,8 +106,16 @@ Holding PHP to avoid upgrading to a newer version without migration..."
...
+if dpkg -l | grep -q mariadb-common
+then
+    print_text_in_color "$ICyan" "MariaDB installed. \
+    Holding MariaDB to avoid crashes..."
+    apt-mark hold mariadb-common
+fi
+
 # Don't allow MySQL/MariaDB
-if ! grep -q pgsql /var/www/nextcloud/config/config.php || dpkg -l | grep -q mariadb-common
+if ! grep -q pgsql /var/www/nextcloud/config/config.php
```
Comment (szaimen) [inline nextcloud_update.sh:118]: would this solve both problems?

Comment (enoch85) [reply nextcloud_update.sh:118]: This doesn't cover SQLite, MySQL or any other DB as it only checks for psql - I think.

If psql is not in config.php AND mariadb exists, THEN = true.

Comment (szaimen) [reply nextcloud_update.sh:118]: Hm...

**Comment by enoch85** [comment]
Let's not do this.

**Pull request closed without merging**